cores/rp2040/SerialUART.cpp

```diff
@@ -365,6 +366,23 @@ SerialUART::operator bool() {
     return _running;
 }
 
+bool SerialUART::getBreakReceived() {
+    CoreMutex m(&_fifoMutex);
+    if (!_running || !m) {
+        return false;
+    }
+
+    if (_polling) {
+        _handleIRQ(false);
+    } else {
+        _pumpFIFO();
+    }
+
+    bool break_received = _break;
```

Comment:
I think the mutex lock needs to be here around only the dangerous bit, and use the raw mutex handler:
````
uint32_t owner;
mutex_enter(&_fifoMutex, &owner);
bool break_received = _break;
_break = false;
mutex_exit(&_fifoMutex);
return break_received;
````

I also think I goofed on my `overflow` implementation due to this same race condition.  I'll fix that after we merge here.  😭  Multicore/thread with shared HW resources is hard!

Reply:
Yeah, it is :)  OK, let me fix that.